src/arthur_common/aggregations/functions/agentic_aggregations.py

```diff
@@ -0,0 +1,727 @@
+            spans_with_metrics_and_agents = extract_spans_with_metrics_and_agents(root_spans)
+            
+            # Process each span with metrics
+            for span, agent_name in spans_with_metrics_and_agents:
+                metric_results = span.get('metric_results', [])
+                
+                for metric_result in metric_results:
+                    metric_type = metric_result.get('metric_type')
+                    details = metric_result.get('details', {})
+                    
+                    if metric_type == 'ToolSelection':
+                        tool_selection = details.get('tool_selection', {})
+                        
+                        # Extract tool selection data
+                        tool_selection_score = tool_selection.get('tool_selection')
+                        tool_selection_reason = tool_selection.get('tool_selection_reason', 'Unknown')
+                        
+                        if tool_selection_score is not None:
+                            tool_selection_data.append({
+                                'ts': ts,
+                                'tool_selection_score': tool_selection_score,
+                                'tool_selection_reason': tool_selection_reason
+                            })
+                        
+                        # Extract tool usage data
+                        tool_usage_score = tool_selection.get('tool_usage')
+                        tool_usage_reason = tool_selection.get('tool_usage_reason', 'Unknown')
+                        
+                        if tool_usage_score is not None:
+                            tool_usage_data.append({
+                                'ts': ts,
+                                'tool_usage_score': tool_usage_score,
+                                'tool_usage_reason': tool_usage_reason
+                            })
+                    
+                    elif metric_type == 'QueryRelevance':
+                        query_relevance = details.get('query_relevance', {})
+                        
+                        # Calculate average relevance score
+                        avg_score = calculate_average_relevance_score(query_relevance)
+                        reason = query_relevance.get('reason', 'Unknown')
+                        
+                        if avg_score is not None:
+                            query_relevance_data.append({
+                                'ts': ts,
+                                'average_relevance_score': avg_score,
+                                'llm_relevance_score': query_relevance.get('llm_relevance_score'),
+                                'reranker_relevance_score': query_relevance.get('reranker_relevance_score'),
+                                'bert_f_score': query_relevance.get('bert_f_score'),
+                                'reason': reason
+                            })
```

Comment:
Similar to above question - is it useful to provide the average of all these values together? (esp. if we are returning all of the scores individually)?